lib/ruby_indexer/lib/ruby_indexer/index.rb

```diff
@@ -129,6 +129,17 @@ def fuzzy_search(query)
       results.flat_map(&:first)
     end
 
+    sig { params(name: String, receiver_name: String).returns(T::Array[Entry]) }
+    def method_completion_candidates(name, receiver_name)
+      ancestors = linearized_ancestors_of(receiver_name)
+      candidates = prefix_search(name)
+      candidates = candidates.flatten
```

Comment:
```suggestion
      candidates.flatten!
```

Reply:
Unfortunately, Sorbet doesn't understand that `flatten!` will change the type of the `candidates` variable from `T::Array[T::Array[String]]` to `T::Array[String]`.

I don't like the style either, but it's better than having to use `T.unsafe`.